plugins/dev-scripts/scripts/test.js

```diff
@@ -30,7 +30,7 @@ console.log(`Building tests for ${packageJson.name}`);
 const args = process.argv.slice(2);
 const skipLint = args.includes('--skip-lint');
 const config = webpackConfig({
-  mode: 'test',
+  mode: 'development',
```

Comment:
I think this change broke all the tests. When running `npm run test` you now get "0 tests passed" for all plugins. I think because of this line https://github.com/BeksOmega/blockly-samples/blob/04004b282c59ed65db0a14e8312303cd61c5d275/plugins/dev-scripts/config/webpack.config.js#L27

without `isTest` being true the entry points for the mocha files never get created.

However just changing this line back to `mode: 'test'` gives me additional errors like

```
ValidationError: Invalid options object. Ignore Plugin has been initialized using an options object that does not match the API schema.
```

that I don't fully understand. I am looking into this issue but I'm wondering if either Beka you could help look into this to fix it more quickly or if we should revert this PR until it's fixed? Because right now all PRs will report that all the tests pass, but in reality it's just not running any tests.

Reply:
Hmmm so I changed this because it seemed like `test` was no longer a valid mode, I thought `development` was the new name (see [the docs](https://webpack.js.org/configuration/mode/) and [the upgrade guide](https://webpack.js.org/migrate/5/#make-sure-to-use-mode)). But honestly it's a bit unclear.

It looks like what Abby is doing is working. But another idea is based on [this old github issue](https://github.com/webpack/webpack/issues/7659), maybe we actually need to be setting [optimization.nodeEnv](https://webpack.js.org/configuration/optimization/#optimizationnodeenv)?

Reply:
I'll have to double check with Sam, but AFAICT 'test' was never a webpack option (even if you go to [webpack 4](https://v4.webpack.js.org/configuration/mode/) that we were using before it does not mention 'test' as an option) , but instead something we set and [use internally](https://github.com/google/blockly-samples/blob/47d85d1e5c53a716692d3091ff3ab39422b020e7/plugins/dev-scripts/config/webpack.config.js#L27).

Again, I'll have to double check. Not pretending to be super confident in my webpack config knowledge here haha.